File: app/src/main/java/com/example/haidermirza/banner/viper/contract/MainContract.java
package com.example.haidermirza.banner.viper.contract;

import android.content.Context;

import com.example.haidermirza.banner.library.BannerLayout;

public interface MainContract {
    interface MainView{
        void show_message(String str);
    }
    interface MainPresenter{
        void attach_view(MainContract.MainView view);
        void on_initial_dataLoad(BannerLayout recyclerBanner, BannerLayout bannerVertical,Context context);
        void on_banner_clik(String path);
        void on_start_normalActivity(Context context);
        void on_start_overFlyingActivity(Context context);
    }
    interface MainInterator{
        void dataLoad(BannerLayout recyclerBanner, BannerLayout bannerVertical,Context context);
        void attach_presenter(MainContract.MainPresenter presenter);
    }
    interface MainRouter{
        void start_normalActivity(Context context);
        void start_overFlyingActivity(Context context);
    }
}

File: app/src/main/java/com/example/haidermirza/banner/viper/contract/MainContract.kt
package com.example.haidermirza.banner.viper.contract

import android.content.Context

import com.example.haidermirza.banner.library.BannerLayout

interface MainContract {
    interface MainView {
        fun show_message(str: String)
    }

    interface MainPresenter {
        fun attach_view(view: MainContract.MainView)
        fun on_initial_dataLoad(recyclerBanner: BannerLayout, bannerVertical: BannerLayout, context: Context)
        fun on_banner_clik(path: String)
        fun on_start_normalActivity(context: Context)
        fun on_start_overFlyingActivity(context: Context)
    }

    interface MainInterator {
        fun dataLoad(recyclerBanner: BannerLayout, bannerVertical: BannerLayout, context: Context)
        fun attach_presenter(presenter: MainContract.MainPresenter)
    }

    interface MainRouter {
        fun start_normalActivity(context: Context)
        fun start_overFlyingActivity(context: Context)
    }
}
